lib/puppet/metatype/manager.rb

```diff
@@ -146,16 +156,19 @@ def type(name)
     # We are overwhelmingly symbols here, which usually match, so it is worth
     # having this special-case to return quickly.  Like, 25K symbols vs. 300
     # strings in this method. --daniel 2012-07-17
-    return @types[name] if @types[name]
+    return @types[name] if @types.include? name
```

Comment:
Trying to set a value here of anything but nil seems to do some bad things that I don't understand yet (it makes a notify resource invalid). So far I've tried: a class called CacheMiss, a string "CacheMiss", a symbol :CacheMiss.